ADR/00xx-related-pipelines.md

```diff
@@ -0,0 +1,104 @@
+
+## Applied to Use Cases
+
+Let's apply the architecture to some use cases, and see how it plays out:
+
+### Common Parent Image
+
+Scenario: an application image depends on a common parent image. The user has 1 Application, and 2 Components. One of them is the parent image. The other image is built FROM that image.
+
+* The common parent image will register a dependency to all other Components
+* After the parent image is built, all other dependent Component images will be rebuilt.
+* If any of these dependent components are further dependent on images (i.e. a bundle image), then those images will also be rebuilt.
+* Intermediate image references can be replaced either by digest or by full image reference as these are used within the same snapshot. The full pull spec should match in any dependent image.
+
+### OLM Operators, with components in different repos
+
+Scenario: the user has 5 components. One of them is a “bundle” image. It contains references to the other four images by digest. None of the images are in the same repo.
+
+* Each non-bundle component declares that it depends on the bundle image.
+* When a change is introduced into any of the non-bundle images, the pullspec will be replaced in the bundle image to enable the change to be tested.
+* Since the pullspec location (i.e. registry and repository) might not be consistent with that in the snapshot due to the fact that the bundle image is targeting the _final_ location of the image, replacement will have to be performed by only finding and replacing the image digests.
+
+### OLM Operators, with components in the same repo
+
+Another OLM Operator Scenario: an operator git repo contains both the controller code and the bundle metadata.
+
+* The bundle component will not indicate any dependency in its PR definition.
+* The operator PR tekton definition will indicate a dependency on the bundle image.
+* The bundle image will be rebuilt after the operator is built to replace the pullspec referenced in the bundle image.
+* Since the pullspec location (i.e. registry and repository) might not be consistent with that in the snapshot due to the fact that the bundle image is targeting the _final_ location of the image, replacement will have to be performed by only finding and replacing the image digests.
+
+## Consequences
+
+* The creation and maintenance of PRs for builds running from AppStudio components remains unchanged (an alternative design that we considered involved creating PRs on other components to maintain the references). This means that the effect on the user managing multiple components in an application is reduced as there are fewer PRs to maintain and keep track of. 
+* The user can pin their digest references in git. AppStudio will automate maintaining them. No magical resolution of tags in the buildsystem at build time, or worse at runtime.
```

Comment:
AppStudio will automate maintaining them. -.> AppStudio will automate their maintenance.
buildsystem -> build system.
That last sentence is missing a verb. Maybe start it with "There is" or "There will be".